iOS/Projects/Features/Record/Sources/Presentation/RecordScene/ViewModel/RecordCalendarViewModel.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  RecordCalendarViewModel.swift
+//  RecordFeature
+//
+//  Created by 안종표 on 11/23/23.
+//  Copyright © 2023 kr.codesquad.boostcamp8. All rights reserved.
+//
+
+import Combine
+import Coordinator
+import Foundation
+
+// MARK: - RecordCalendarViewModelInput
+
+struct RecordCalendarViewModelInput {
+  let appear: AnyPublisher<Void, Never>
+  let appearSection: AnyPublisher<Int, Never>
+  let calendarDateDidTapped: AnyPublisher<IndexPath, Never>
+  let calendarCellReuse: AnyPublisher<Void, Never>
+}
+
+typealias RecordCalendarViewModelOutput = AnyPublisher<RecordCalendarState, Error>
+
+// MARK: - RecordCalendarState
+
+enum RecordCalendarState {
+  case date([DateInfo])
+  case selectedIndexPath(IndexPath)
+}
+
+// MARK: - RecordCalendarViewModel
+
+final class RecordCalendarViewModel {
+  private var subscriptions: Set<AnyCancellable> = []
+  private let dateProvideUseCase: DateProvideUseCaseRepresentable
+
+  private(set) var currentSelectedIndexPath: IndexPath?
+
+  init(dateProvideUseCase: DateProvideUseCaseRepresentable) {
+    self.dateProvideUseCase = dateProvideUseCase
+  }
+}
+
+// MARK: RecordCalendarViewModelRepresentable
+
+extension RecordCalendarViewModel: RecordCalendarViewModelRepresentable {
+  func transform(input: RecordCalendarViewModelInput) -> RecordCalendarViewModelOutput {
+    for subscription in subscriptions {
+      subscription.cancel()
+    }
+    subscriptions.removeAll()
+
+    let appearTotalDateInfo = input.appear
+      .flatMap { [weak self] _ -> AnyPublisher<[DateInfo], Error> in
+        guard let self else {
+          return Fail(error: BindingError.viewModelDeinitialized)
+            .eraseToAnyPublisher()
+        }
+        let allDates = dateProvideUseCase.fetchAllDatesThisMonth()
+        return Just(allDates)
+          .setFailureType(to: Error.self)
+          .eraseToAnyPublisher()
+      }
+      .map { dateInfos -> RecordCalendarState in
+        .date(dateInfos)
+      }
+      .eraseToAnyPublisher()
```

Comment:
p3: flatMap을 이용할 때 단순히 함수만을 호출해서 처리해도 좋을 것 같아요.
```suggestion
    let appearTotalDateInfo = input.appear
      .flatMap(dateProvideUseCase.fetchAllDatesThisMonth)
      .map { dateInfos -> RecordCalendarState in
        .date(dateInfos)
      }
      .eraseToAnyPublisher()
```